### PR TITLE
Making naming of PubSub resources consistent and more readable

### PR DESCRIPTION
**Pull request by nachocano**
Fixes https://github.com/google/knative-gcp/issues/1155

## Proposed Changes

- Using the convention used in the broker: `cre-<owner_type>_<namespace>_<name>_<uid>`.
- Moving things around to avoid duplication.
- Changing PS's receive adapter name. This was also hard to debug. This will also be breaking and we should probably provide a post-install script in OSS to delete old deployments.
- Adding UTs.
- CAL sink size limit (100) is different than regular PubSub resources (255). Setting it to that new maximum as it was causing problems in PROW as we are using very long namespaces and names..

**Release Note**

```release-note
We have changed the naming convention we use for GCP resources in order to make it easy for users to understand who created them (e.g., a particular source, a channel?). The convention is `cre-<owner_type>_<namespace>_<name>_<uid>`. For example if a Source mysource in the namespace default with uid 47163a creates a Pub/Sub subscription, then the subscription will be called `cre-src_default_mysource_47163a`. We highly recommend users when upgrading to 0.16 to delete ALL their sources and channels first so that the previous Pub/Sub resources are removed, OR to delete the GCP resources using our post-installation script. 
We also changed the name of the PullSubscription's receive adapters. We will provide a post-installation script that will delete previous deployments.
```

### CODE REVIEW

**Comment by nachocano** [comment]
/assign @Harwayne 
fyi @zargarpur

**Comment by nachocano** [comment]
@Harwayne this is ready for review...

File: pkg/reconciler/events/auditlogs/resources/names.go [hunk 1]
```diff
@@ -38,5 +39,6 @@ func GenerateTopicResourceName(s *v1alpha1.CloudAuditLogsSource) string {
 // GenerateSinkName generates a Stackdriver sink resource name for an
 // CloudAuditLogsSource.
 func GenerateSinkName(s *v1alpha1.CloudAuditLogsSource) string {
-	return fmt.Sprintf("sink-%s", string(s.UID))
+	// This name should be different than the one provided by GenerateTopicName.
```
Comment (Harwayne) [inline pkg/reconciler/events/auditlogs/resources/names.go:42]: Add to this comment saying why it has to be different.

Comment (nachocano) [reply pkg/reconciler/events/auditlogs/resources/names.go:42]: I may have been wrong. The problem might have been just about the length. Let me try again with `cre-src`... If it doesn't work, then I'll switch back to `cre-cal` and add the comment

**Comment by nachocano** [comment]
Created an issue to track the proper removal of resources... https://github.com/google/knative-gcp/issues/1217

**Comment by nachocano** [comment]
seeing

```2020/06/04 19:53:37 Unexpected error running "go build": signal: killed
2020/06/04 19:53:37 Unexpected error running "go build": signal: killed
2020/06/04 19:53:37 Unexpected error running "go build": signal: killed
2020/06/04 19:53:37 error processing import paths in "/home/prow/go/src/github.com/google/knative-gcp/test/test_images/auditlogs_receiver/pod.yaml": error resolving image references: exit status 1```

**Comment by Harwayne** [comment]
> seeing
> 
> ```
> 2020/06/04 19:53:37 Unexpected error running "go build": signal: killed
> 2020/06/04 19:53:37 Unexpected error running "go build": signal: killed
> 2020/06/04 19:53:37 error processing import paths in "/home/prow/go/src/github.com/google/knative-gcp/test/test_images/auditlogs_receiver/pod.yaml": error resolving image references: exit status 1```
> ```

```
error processing import paths in "/home/prow/go/src/github.com/google/knative-gcp/test/test_images/auditlogs_receiver/pod.yaml": error resolving image references: exit status 1
```

I was seeing the same thing in #1156.

**Comment by nachocano** [comment]
> > seeing
> > ```
> > 2020/06/04 19:53:37 Unexpected error running "go build": signal: killed
> > 2020/06/04 19:53:37 Unexpected error running "go build": signal: killed
> > 2020/06/04 19:53:37 error processing import paths in "/home/prow/go/src/github.com/google/knative-gcp/test/test_images/auditlogs_receiver/pod.yaml": error resolving image references: exit status 1```
> > ```
> 
> ```
> error processing import paths in "/home/prow/go/src/github.com/google/knative-gcp/test/test_images/auditlogs_receiver/pod.yaml": error resolving image references: exit status 1
> ```
> 
> I was seeing the same thing in #1156.

ok, yeah, might be something weird with PROW. Hope the retest will pass.
@chizhg something that might be worth looking if you have any cycles left

**Comment by nachocano** [comment]
another one 

```
2020/06/04 20:52:00 Unexpected error running "go build": signal: killed
2020/06/04 20:52:00 error processing import paths in "/home/prow/go/src/github.com/google/knative-gcp/test/test_images/pubsub_target/pod.yaml": error resolving image references: exit status 1
```

/retest

**Comment by chizhg** [comment]
> > > seeing
> > > ```
> > > 2020/06/04 19:53:37 Unexpected error running "go build": signal: killed
> > > 2020/06/04 19:53:37 Unexpected error running "go build": signal: killed
> > > 2020/06/04 19:53:37 error processing import paths in "/home/prow/go/src/github.com/google/knative-gcp/test/test_images/auditlogs_receiver/pod.yaml": error resolving image references: exit status 1```
> > > ```
> > 
> > 
> > ```
> > error processing import paths in "/home/prow/go/src/github.com/google/knative-gcp/test/test_images/auditlogs_receiver/pod.yaml": error resolving image references: exit status 1
> > ```
> > 
> > 
> > I was seeing the same thing in #1156.
> 
> ok, yeah, might be something weird with PROW. Hope the retest will pass.
> @chizhg something that might be worth looking if you have any cycles left

Seems to be the OOM issue we had for build tests? We haven't seen integration test is taking so much memory for other repos so far, but I think making changes to the knative-gcp integration test Prow jobs (similar as done by https://github.com/knative/test-infra/pull/2146) would fix this issue.

**Comment by nachocano** [comment]
> > > > seeing
> > > > ```
> > > > 2020/06/04 19:53:37 Unexpected error running "go build": signal: killed
> > > > 2020/06/04 19:53:37 Unexpected error running "go build": signal: killed
> > > > 2020/06/04 19:53:37 error processing import paths in "/home/prow/go/src/github.com/google/knative-gcp/test/test_images/auditlogs_receiver/pod.yaml": error resolving image references: exit status 1```
> > > > ```
> > > 
> > > 
> > > ```
> > > error processing import paths in "/home/prow/go/src/github.com/google/knative-gcp/test/test_images/auditlogs_receiver/pod.yaml": error resolving image references: exit status 1
> > > ```
> > > 
> > > 
> > > I was seeing the same thing in #1156.
> > 
> > 
> > ok, yeah, might be something weird with PROW. Hope the retest will pass.
> > @chizhg something that might be worth looking if you have any cycles left
> 
> Seems to be the OOM issue we had for build tests? We haven't seen integration test is taking so much memory for other repos so far, but I think making changes to the knative-gcp integration test Prow jobs (similar as done by [knative/test-infra#2146](https://github.com/knative/test-infra/pull/2146)) would fix this issue.

@capri-xiyue I need to step out for a few hours. Can you help me out and increase those limits so that we can get this PR in? If you don't have cycles, no worries, I'll do once I'm back

**Comment by capri-xiyue** [comment]
Submitted this PR https://github.com/knative/test-infra/pull/2166 to increase the mem limit of prow e2e test jobs

**Comment by nachocano** [comment]
YESSSSSSSSSSSSSSS!!!! FINALLYYYY

Thanks @capri-xiyue and @Harwayne
